From ee63503d48cc1e82bc0ff8a6fe5918fa699dd8e1 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Tue, 06 Apr 2021 18:13:56 +0300
Subject: [PATCH] [SW-37437] gaudi: ppboot: add pci device id

host must be able to tell if device in
secure mode only by picking in lspci.
This will serve LKD as we are going
to protect ELBI access.

Hence 2 new pci device ids have been
configured as:
Gaudi secure :	  0x1010
Gaudi FP16 secure:0x1011

Change-Id: I89ac4cb955a11b873330b1d1807f19801f9f5815
Signed-off-by: Karthikeyan K <karthikk@habana.ai>
---

diff --git a/gaudi/Makefile b/gaudi/Makefile
index e9fe78b..bc21aee 100644
--- a/gaudi/Makefile
+++ b/gaudi/Makefile
@@ -7,6 +7,8 @@
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/reset.c)
 PREBOOT_SRCS_C += $(shell ls $(HL_DEV)/rma_info.c)
 
+PPBOOT_SRCS_C += $(shell ls $(HL_DEV)/security.c)
+
 CFLAGS += -mstrict-align
 
 image_validation: image_validation_common
diff --git a/gaudi/common.c b/gaudi/common.c
index 21c008b..7a55981 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -110,6 +110,7 @@
 }
 
 #define FUSE_DEV_ID_WORD_OFFSET	1
+#define GAUDI_SECURE_PCI_ID_EXT	BIT(4)
 
 u32 preboot_get_vendor_id_device_id(void)
 {
@@ -124,6 +125,15 @@
 			pci_dev_id = PCI_IDS_GAUDI_HL2000M;
 	}
 
+	/*
+	 * Secure pci ids:
+	 * PCI_IDS_GAUDI_SEC : 0x1010
+	 * PCI_IDS_GAUDI_HL2000M_SEC : 0x1011
+	 */
+	/* If security is enabled set the PCI ID accordingly*/
+	if (preboot_security_is_enabled())
+		pci_dev_id |= GAUDI_SECURE_PCI_ID_EXT;
+
 	return (pci_dev_id << 16) | PCI_VENDOR_ID_HABANALABS;
 }
 
